waspc/data/Generator/templates/sdk/wasp/server/config.ts

```diff
@@ -1,94 +1,47 @@
 {{={= =}=}}
-import merge from 'lodash.merge'
+import { env } from './env.js'
+import { stripTrailingSlash } from '../universal/url.js'
 
-import { stripTrailingSlash } from "../universal/url.js";
+type NodeEnv = typeof env.NODE_ENV
 
-const nodeEnv = process.env.NODE_ENV ?? 'development'
-
-// TODO:
-//   - Use dotenv library to consume env vars from a file.
-//   - Use convict library to define schema and validate env vars.
-//  https://codingsans.com/blog/node-config-best-practices
-
-type BaseConfig = {
+type Config = {
+  env: NodeEnv;
+  isDevelopment: boolean;
+  port: number;
+  databaseUrl: string;
+  frontendUrl: string;
+  serverUrl: string;
   allowedCORSOrigins: string | string[];
   {=# isAuthEnabled =}
   auth: {
-    jwtSecret: string | undefined;
+    jwtSecret: string;
   }
   {=/ isAuthEnabled =}
 }
 
-type CommonConfig = BaseConfig & {
-  env: string;
-  isDevelopment: boolean;
-  port: number;
-  databaseUrl: string | undefined;
-}
+const frontendUrl = stripTrailingSlash(env.WASP_WEB_CLIENT_URL)
+const serverUrl = stripTrailingSlash(env.WASP_SERVER_URL)
 
-type EnvConfig = BaseConfig & {
-  frontendUrl: string;
-  serverUrl: string;
+const allowedCORSOriginsPerEnv: Record<NodeEnv, string | string[]> = {
+  development: '*',
+  production: [frontendUrl]
 }
-
-type Config = CommonConfig & EnvConfig
-
-const config: {
-  all: CommonConfig,
-  development: EnvConfig,
-  production: EnvConfig,
-} = {
-  all: {
-    env: nodeEnv,
-    isDevelopment: nodeEnv === 'development',
-    port: process.env.PORT ? parseInt(process.env.PORT) : {= defaultServerPort =},
-    databaseUrl: process.env.{= databaseUrlEnvVarName =},
-    allowedCORSOrigins: [],
-    {=# isAuthEnabled =}
-    auth: {
-      jwtSecret: undefined
-    }
-    {=/ isAuthEnabled =}
-  },
-  development: getDevelopmentConfig(),
-  production: getProductionConfig(),
-}
-
-const resolvedConfig: Config = merge(config.all, config[nodeEnv])
-// PUBLIC API
-export default resolvedConfig
-
-function getDevelopmentConfig(): EnvConfig {
-  const frontendUrl = stripTrailingSlash(process.env.WASP_WEB_CLIENT_URL ?? '{= defaultClientUrl =}');
-  const serverUrl = stripTrailingSlash(process.env.WASP_SERVER_URL ?? '{= defaultServerUrl =}');
-  return {
-    // @ts-ignore
-    frontendUrl,
-    // @ts-ignore
-    serverUrl,
-    allowedCORSOrigins: '*',
-    {=# isAuthEnabled =}
-    auth: {
-      jwtSecret: 'DEVJWTSECRET'
-    }
-    {=/ isAuthEnabled =}
+const allowedCORSOrigins = allowedCORSOriginsPerEnv[env.NODE_ENV]
+
+const config: Config = {
```

Comment:
Simplified the creation of the `config` object, instead of having three separate things `all`, `development`, `production`, now we construct a single object and vary some of the values (e.g. allowed CORS origins) based on `NODE_ENV`.

Reply:
I think the previous approach we clever since it envisioned many different options for dev and for production, but it in the end, we only vary the CORS origins between dev and production. So, it seemed like an overkill and I know I had these complex types to capture the different objects. But when I looked closer, it turned out this simple `Config` type is enough.

Reply:
Ok, makes sense, we can always modify it if needed!